src/components/Package.tsx

```diff
@@ -13,43 +17,102 @@ interface PackageState {
+            ))
+                .reduce(flatten, [])
+                .sort((a, b) => a.date < b.date ? 1 : -1);
+            state.versionsData = versionsData;
 
-        const versionsData = (await Promise.all(
-            packageData.map(({ version }) => getJSON<any[]>(`${config.apiRoot}/status/${packageName}/${version}.json`))
-        ))
-            .reduce(flatten, [])
-            .sort((a, b) => a.date < b.date ? 1 : -1);
-        this.state.versionsData = versionsData;
+            state.versionsCardData = await Promise.all(
+                packageData.map((element) => {
+                    return getJSON<any[]>(`${config.apiRoot}/status/${packageName}/${element.version}.json`)
+                        .then((result) => {
+                            return {
+                                ...element,
+                                detail: result
+                            }
+                        });
+                })
+            );
+        }
+        catch{
+            console.log('We failed')
+            state.packageNotFound = true
+            console.log(state.packageNotFound)
+        }
     }
 
     render() {
-        if (!this.state.versionsData) {
-            return <h1>Loading...</h1>
+        if(state.pageURL != window.location.href){
+            state.pageURL = window.location.href;
+            this.updatePackage()
+        }
+        if (!state.versionsData) {
+            return(
+                <> 
+                    <h1>Loading...</h1>
+                    {state.packageNotFound == true &&
+                        <>
+                        {console.log('Yo')}
+                        <Redirect to='/packagenotfound/'></Redirect>
+                        </>
+                    }
+                </>
+            )
         } else {
             return <div>
-                <h1>Package {this.state.packageName}!</h1>
-                <pre>{JSON.stringify(this.state.packageData, null, '\t')}</pre>
-                <pre>{JSON.stringify(this.state.versionsData, null, '\t')}</pre>
+                {state.packageNotFound == true &&
+                    <Redirect to='/packagenotfound/'></Redirect>
+                }
+                {state.versionsCardData != undefined &&
+                    <Header title={state.packageName} package={true} version={state.versionsCardData[0].version} grade={state.versionsCardData[0].grade}></Header>
+                }
+                <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center',height:'82vh',overflowY:'scroll'}}>
+                    <div style={{ display: 'flex', flexDirection: 'column', width: '90%',paddingTop:'10px'}}>
+                        {state.versionsCardData != undefined &&
+                            state.versionsCardData.map((entity: any, index: number) =>
+                                <div className={(index % 2 == 0 ? 'pull-left' : 'pull-right')} style={{width:'40vw'}}>
```

Comment:
Move this `index % 2 == 0` logic into a variable in the render method, rather than in-lining it